pkg/server/delta/v3/server.go

```diff
@@ -44,6 +132,340 @@ func NewServer(ctx context.Context, config cache.ConfigWatcher, callbacks Callba
+
+	for {
+		select {
+		case <-s.ctx.Done():
+			return nil
+			// config watcher can send the requested resources types in any order
+		case resp, more := <-values.deltaEndpoints:
+			if !more {
+				return status.Errorf(codes.Unavailable, "endpoints watch failed")
+			}
+			nonce, err := send(resp, resource.EndpointType)
+			if err != nil {
+				return err
+			}
+			values.deltaEndpointNonce = nonce
+			setState(resp, resource.EndpointType)
+		case resp, more := <-values.deltaClusters:
+			if !more {
+				return status.Errorf(codes.Unavailable, "clusters watch failed")
+			}
+			nonce, err := send(resp, resource.ClusterType)
+			if err != nil {
+				return err
+			}
+			values.deltaClusterNonce = nonce
+			setState(resp, resource.ClusterType)
+		case resp, more := <-values.deltaRoutes:
+			if !more {
+				return status.Errorf(codes.Unavailable, "routes watch failed")
+			}
+			nonce, err := send(resp, resource.RouteType)
+			if err != nil {
+				return err
+			}
+			values.deltaRouteNonce = nonce
+			setState(resp, resource.RouteType)
+		case resp, more := <-values.deltaListeners:
+			if !more {
+				return status.Errorf(codes.Unavailable, "listeners watch failed")
+			}
+			nonce, err := send(resp, resource.ListenerType)
+			if err != nil {
+				return err
+			}
+			values.deltaListenerNonce = nonce
+			setState(resp, resource.ListenerType)
+		case resp, more := <-values.deltaSecrets:
+			if !more {
+				return status.Errorf(codes.Unavailable, "secrets watch failed")
+			}
+			nonce, err := send(resp, resource.SecretType)
+			if err != nil {
+				return err
+			}
+			values.deltaSecretNonce = nonce
+			setState(resp, resource.SecretType)
+		case resp, more := <-values.deltaRuntimes:
+			if !more {
+				return status.Errorf(codes.Unavailable, "runtimes watch failed")
+			}
+			nonce, err := send(resp, resource.RuntimeType)
+			if err != nil {
+				return err
+			}
+			values.deltaRuntimeNonce = nonce
+			setState(resp, resource.RuntimeType)
+		case resp, more := <-values.deltaResponses:
+			if more {
+				if resp == deltaErrorResponse {
+					return status.Errorf(codes.Unavailable, "delta resource watch failed")
+				}
+				typeURL := resp.GetDeltaRequest().TypeUrl
+				nonce, err := send(resp, typeURL)
+				if err != nil {
+					return err
+				}
+				values.deltaNonces[typeURL] = nonce
+				setState(resp, typeURL)
+			}
+		case req, more := <-reqCh:
+			// input stream ended or errored out
+			if !more {
+				return nil
+			}
+			if req == nil {
+				return status.Errorf(codes.Unavailable, "empty request")
+			}
+
+			// Log out our error detail from  envoyif we get one but don't do anything crazy here yet
+			// TODO: embed a logger in the server so we can be a bit more verbose when needed
+			if req.ErrorDetail != nil {
+				return status.Errorf(codes.Code(req.ErrorDetail.GetCode()), "received error from xDS client: %s", req.ErrorDetail.GetMessage())
+			}
+
+			// node field in discovery request is delta-compressed
```

Comment:
Elaborate on what delta-compressed means or link out

Reply:
Honestly I have no clue, that's another SOTW copy. I removed that comment